mlir/Makefile

```diff
@@ -48,6 +48,8 @@ llvm:
 
 .PHONY: mhlo
 mhlo:
+	patch -d mlir-hlo/mhlo/analysis --follow-symlinks < patches/add-lmhlo-dependency.patch
+
```

Comment:
Oh wow, two new patches 😅 Would you mind explaining what they do and why?

Reply:
>One is for this https://github.com/tensorflow/mlir-hlo/issues/68 issue. There's an error in mlir-hlo's build system where a dependency is not stated. The error would only show up under certain situations for instance, building in systems with only one processor that forces CMake to execute in a certain order. It doesn't show up when compiling in parallel with multiple processors reliably. Since we compile in github actions, with limited number of processors, the build process fails.

Oh nice good find. We should add a TODO comment here to remove the line once the issue is resolved.

>The other is for removing some MLIR python binding generated code that is repeated in the JAX distribution. During the loading of the module, python complains that these definitions were already loaded.

Got it 👍

Reply:
https://github.com/PennyLaneAI/catalyst/pull/143/commits/7db5fc095a8afb5d1b372e009ef3e2668598b704